assets/js/components/KeyMetrics/ChipTabGroup/index.js

```diff
@@ -153,6 +181,25 @@ export default function ChipTabGroup( { allMetricItems, savedItemSlugs } ) {
 			).length;
 			selectedCounts[ metricGroup ] = selectedCount;
 		}
+
+		// Check if metric is conversion event related and if new badge should be included.
+		if ( conversionReportingEventsChange?.newEvents ) {
+			const isNewlyDetectedKeyMetrics =
+				conversionReportingEventsChange.newEvents.some(
+					( conversionEvent ) =>
+						conversionReportingEventWidgets[
+							conversionEvent
+						].includes( metricItemSlug )
+				);
+
+			if ( isNewlyDetectedKeyMetrics ) {
+				newlyDetectedMetrics[ metricGroup ] = newlyDetectedMetrics[
+					metricGroup
+				]
+					? [ ...newlyDetectedMetrics[ metricGroup ], metricItemSlug ]
+					: [ metricItemSlug ];
+			}
```

Comment:
This is pretty tough to read, but probably a bit unnecessary:

```suggestion
			if ( isNewlyDetectedKeyMetrics ) {
				newlyDetectedMetrics[ metricGroup ] = [ ...(newlyDetectedMetrics[ metricGroup ] ?? []), metricItemSlug ];
			}
```